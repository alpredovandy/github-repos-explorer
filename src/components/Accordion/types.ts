import React from 'react';

export interface AccordionProps {
    title: string;
    content: React.ReactNode;
}
