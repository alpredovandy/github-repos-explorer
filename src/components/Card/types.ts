export interface CardProps {
    data: Record<string, any>;
    isLoading?: boolean;
}
