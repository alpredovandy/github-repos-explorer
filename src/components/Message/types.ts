export interface MessageProps {
    show: boolean;
    onClose: () => void;
    delay?: number;
    message: string;
    type: 'success' | 'error' | 'info';
}
