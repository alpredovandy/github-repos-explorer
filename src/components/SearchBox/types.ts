export interface SearchBoxProps {
    handleSearch: (search: string) => void;
    handleClear: () => void;
    value: string;
    placeholder?: string;
}
