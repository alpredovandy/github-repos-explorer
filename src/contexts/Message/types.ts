export interface MessageData {
    type?: 'success' | 'error' | 'info';
    message?: string;
}

export interface MessageState {
    /** open message */
    show: boolean;

    /** message data */
    data?: MessageData;
}

export interface MessageContextProps {
    showMessage: (props: MessageData) => void;
    showSuccessMessage: (props: Omit<MessageData, 'type'>) => void;
    showErrorMessage: (props: Omit<MessageData, 'type'>) => void;
    closeMessage: () => void;
}
