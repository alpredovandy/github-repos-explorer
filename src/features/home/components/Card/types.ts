type Repository = {
    name: string;
    stargazers_count: number;
    description: string;
};

type CardData = {
    username?: string;
    repositories?: Repository[];
};

export interface CardProps {
    data: CardData;
    isLoading?: boolean;
}
