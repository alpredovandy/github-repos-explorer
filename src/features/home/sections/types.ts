export interface CatalogProps {
    datas: Record<string, unknown>[];
    isPending: boolean;
}
