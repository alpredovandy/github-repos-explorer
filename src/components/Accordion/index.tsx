import React, { useState } from 'react';

import { AccordionProps } from './types';

const Accordion: React.FC<AccordionProps> = ({ title, content }) => {
    const [isOpen, setIsOpen] = useState(false);

    const toggleAccordion = () => {
        setIsOpen(!isOpen);
    };

    return (
        <div className="rounded-sm flex flex-col items-end mb-4">
            <button onClick={toggleAccordion} className="flex justify-between items-center w-full px-4 py-2 bg-[#f2f2f2] hover:bg-gray-100">
                <span className="font-medium text-[#2a2a2a]">{title}</span>
                {/* Arrow icons */}
                {isOpen ? (
                    // Up arrow
                    <svg className="w-4 h-4 transition-transform duration-200" fill="none" stroke="currentColor" strokeWidth={2} viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" d="M5 15l7-7 7 7" />
                    </svg>
                ) : (
                    // Down arrow
                    <svg className="w-4 h-4 transition-transform duration-200" fill="none" stroke="currentColor" strokeWidth={2} viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" d="M19 9l-7 7-7-7" />
                    </svg>
                )}
            </button>
            {isOpen && <div className="mt-2 py-2 w-[90%] mb-8">{content}</div>}
        </div>
    );
};

export default Accordion;
