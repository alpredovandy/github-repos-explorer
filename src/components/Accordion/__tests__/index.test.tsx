import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import Accordion from '@/components/Accordion';

import { AccordionProps } from '../types';

describe('Test components: Accordion', () => {
    const props: AccordionProps = {
        title: 'Test Title',
        content: <div>Test Content</div>,
    };

    const renderAccordion = () => render(<Accordion {...props} />);

    it('Snapshot', () => {
        const { container } = renderAccordion();

        expect(container).toMatchSnapshot();
    });
});
