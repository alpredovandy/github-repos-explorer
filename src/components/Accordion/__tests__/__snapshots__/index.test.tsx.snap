// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test components: Accordion > Snapshot 1`] = `
<div>
  <div
    class="rounded-sm flex flex-col items-end mb-4"
  >
    <button
      class="flex justify-between items-center w-full px-4 py-2 bg-[#f2f2f2] hover:bg-gray-100"
    >
      <span
        class="font-medium text-[#2a2a2a]"
      >
        Test Title
      </span>
      <svg
        class="w-4 h-4 transition-transform duration-200"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 9l-7 7-7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</div>
`;
